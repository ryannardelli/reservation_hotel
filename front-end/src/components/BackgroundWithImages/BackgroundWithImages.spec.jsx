import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { BackgroundWithImages } from '.';

describe('BackgroundWithImages Component', () => {
  const backgroundImage = 'https://example.com/background.jpg';
  const images = [
    'https://example.com/image1.jpg',
    'https://example.com/image2.jpg',
  ];

  test('should render background image correctly', () => {
    render(
      <BackgroundWithImages
        backgroundImage={backgroundImage}
        images={images}
      />,
    );

    const backgroundImageElement = screen.getByAltText('background');
    expect(backgroundImageElement).toHaveAttribute('src', backgroundImage);
    expect(backgroundImageElement).toBeInTheDocument();
  });

  test('should render overlay images corretly', () => {
    render(
      <BackgroundWithImages
        backgroundImage={backgroundImage}
        images={images}
      />,
    );
    images.forEach((image, index) => {
      const overlayImage = screen.getByAltText(`Overlay ${index + 1}`);
      expect(overlayImage).toHaveAttribute('src', image);
      expect(overlayImage).toBeInTheDocument();
    });
  });

  test('should render the correct alt text for background and overlay images', () => {
    render(
      <BackgroundWithImages
        backgroundImage={backgroundImage}
        images={images}
      />,
    );
    const backgroundImageElement = screen.getByAltText('background');
    expect(backgroundImageElement).toHaveAttribute('alt', 'background');

    images.forEach((_, index) => {
      const overlayImage = screen.getByAltText(`Overlay ${index + 1}`);
      expect(overlayImage).toHaveAttribute('alt', `Overlay ${index + 1}`);
    });
  });

  test('matches snapshot', () => {
    const { asFragment } = render(
      <BackgroundWithImages
        backgroundImage={backgroundImage}
        images={images}
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
