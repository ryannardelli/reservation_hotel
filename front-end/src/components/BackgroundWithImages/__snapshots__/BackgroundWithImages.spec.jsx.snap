// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackgroundWithImages Component matches snapshot 1`] = `
<DocumentFragment>
  <img
    alt="background"
    class="sc-blHHSb cXzvFF"
    src="https://example.com/background.jpg"
  />
  <div
    class="sc-gtLWhw hXGzWP"
  >
    <img
      alt="Overlay 1"
      class="sc-egkSDF jUQtYF"
      src="https://example.com/image1.jpg"
    />
    <img
      alt="Overlay 2"
      class="sc-egkSDF jUQtYF"
      src="https://example.com/image2.jpg"
    />
  </div>
</DocumentFragment>
`;
